### PR TITLE
NP-172: Added variable load to the requests.

### PR DESCRIPTION
**Pull request by schemabotview**

### CODE REVIEW

File: src/test/scala/uk/gov/hmrc/perftests/example/UknwAuthCheckerApiRequests.scala [hunk 1]
```diff
@@ -20,27 +20,25 @@ import io.gatling.core.Predef._
...
   val baseUrl: String             = baseUrlFor("uknw-auth-checker-api")
   val route: String               = "/authorisations"
   private val bearerToken: String = if (runLocal) s"$${accessToken}" else s"Bearer $${accessToken}"
-
-  val navigateToHomePage: HttpRequestBuilder =
-    http("Navigate to Home Page")
-      .get(s"$baseUrl$route/vat-return-period")
-      .check(status.is(200))
-      .check(css("input[name=csrfToken]", "value").saveAs("csrfToken"))
+  val eoriGen: Gen[String]        = RegexpGen.from("^(GB|XI)[0-9]{12}|(GB|XI)[0-9]{15}$")
```
Comment (TheDanielDoyle) [inline src/test/scala/uk/gov/hmrc/perftests/example/UknwAuthCheckerApiRequests.scala]: This won't work. We have dedicated stub records we allow in our service. Coordinate with Alex to get one that can work with the happy path.

We don't need **n** number of eoris. Just a single eori happy path request will do. It's up to gatlin to manage the load.

Comment (schemabotview) [reply src/test/scala/uk/gov/hmrc/perftests/example/UknwAuthCheckerApiRequests.scala]: Refactored the code to have payload as static data.

File: src/test/resources/journeys.conf [hunk 2]
```diff
@@ -14,18 +14,39 @@
...
-      UknwAuthCheckerApiJourney
-    ]
+  uknw-auth-checker-single-eori = {
+    description = "UKNW Auth Checker Single EORI Journey"
+    load = 0.001
+    parts = [SingleEoriJourney]
   }
+  uknw-auth-checker-100-eori = {
+      description = "UKNW Auth Checker 100 EORI Journey"
+      load = 0.001
```
Comment (TheDanielDoyle) [inline src/test/resources/journeys.conf]: Please can you update the load to what's described per journey on here: https://confluence.tools.tax.service.gov.uk/display/NOP/Performance+testing

File: src/test/resources/services.conf [hunk 3]
```diff
@@ -32,13 +32,13 @@
 services {
     auth-login-api {
       protocol = http
-#     host = **place holder **
+      host = www.staging.tax.service.gov.uk
```
Comment (TheDanielDoyle) [inline src/test/resources/services.conf]: This url is wrong. Should be auth-login-api.protected.mdtp

File: src/test/resources/data/500Eori.json [hunk 4]
```diff
@@ -0,0 +1,5 @@
...
...63476179","XI270463476179","XI270463476179","XI270463476179","XI270463476179","XI270463476179","XI270463476179","XI270463476179","XI270463476179","XI270463476179","XI270463476179","XI270463476179","XI270463476179","XI270463476179","XI270463476179","XI270463476179","XI270463476179","XI270463476179","XI270463476179","XI270463476179","XI270463476179","XI270463476179","XI270463476179","XI270463476179","XI270463476179","XI270463476179","XI270463476179","XI270463476179","XI270463476179","XI270463476179","XI270463476179","XI270463476179","XI270463476179","XI270463476179","XI270463476179","XI270463476179","XI270463476179","XI270463476179","XI270463476179","XI270463476179","XI270463476179","XI270463476179","XI270463476179","XI270463476179","XI270463476179","XI270463476179","XI270463476179","XI270463476179","XI270463476179","XI270463476179","XI270463476179","XI270463476179","XI270463476179","XI270463476179","XI270463476179","XI270463476179","XI270463476179","XI270463476179","XI270463476179","XI270463476179","XI270463476179","XI270463476179","XI270463476179","XI270463476179","XI270463476179","XI270463476179","XI270463476179","XI270463476179","XI270463476179","XI270463476179","XI270463476179"]
```
Comment (TheDanielDoyle) [inline src/test/resources/data/500Eori.json]: Please can you run these through a formatter.

File: src/test/resources/data/100Eori.json [hunk 5]
```diff
@@ -0,0 +1,6 @@
...
...,"GB974163550126687","GB974163550126687","GB974163550126687","GB974163550126687","GB974163550126687","GB974163550126687","GB974163550126687","GB974163550126687","GB974163550126687","GB974163550126687","GB974163550126687","GB974163550126687","GB974163550126687","GB974163550126687","GB974163550126687","GB974163550126687","GB974163550126687","GB974163550126687","GB974163550126687","GB974163550126687","GB974163550126687","GB974163550126687","GB974163550126687","GB974163550126687","GB974163550126687","GB974163550126687","GB974163550126687","GB974163550126687","GB974163550126687","GB974163550126687","GB974163550126687","GB974163550126687","GB974163550126687","GB974163550126687","GB974163550126687","GB974163550126687","GB974163550126687","GB974163550126687","GB974163550126687","GB974163550126687","GB974163550126687","GB974163550126687","GB974163550126687","GB974163550126687","GB974163550126687","GB974163550126687","GB974163550126687","GB974163550126687","GB974163550126687","GB974163550126687","GB974163550126687","GB974163550126687","GB974163550126687","GB974163550126687","GB974163550126687","GB974163550126687","GB974163550126687","GB974163550126687","GB974163550126687","GB974163550126687"
```
Comment (TheDanielDoyle) [inline src/test/resources/data/100Eori.json]: Please can you run these through a formatter.

File: src/test/scala/uk/gov/hmrc/perftests/uknwauthcheckerapi/requests/UknwAuthCheckerApiRequests.scala [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+import io.gatling.http.request.builder.HttpRequestBuilder
+import io.netty.handler.codec.http.HttpResponseStatus
+import uk.gov.hmrc.performance.conf.ServicesConfiguration
+import uk.gov.hmrc.perftests.uknwauthcheckerapi.util.Helper
+
+object UknwAuthCheckerApiRequests extends ServicesConfiguration {
+
+  val baseUrl: String             = baseUrlFor("uknw-auth-checker-api")
+  val route: String               = "/authorisations"
+  private val bearerToken: String = if (runLocal) s"$${accessToken}" else s"Bearer $${accessToken}"
```
Comment (TheDanielDoyle) [inline src/test/scala/uk/gov/hmrc/perftests/uknwauthcheckerapi/requests/UknwAuthCheckerApiRequests.scala]: How come you have double $$?

File: src/test/scala/uk/gov/hmrc/perftests/uknwauthcheckerapi/requests/UknwAuthCheckerApiRequests.scala [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+
+  val postAuthorisation: HttpRequestBuilder =
+    http("Post Authorisations")
+      .post(s"$baseUrl$route": String)
+      .body(StringBody(Helper.singleEoriJsonBody))
+      .headers(
+        Map(
+          HttpHeaderNames.Authorization -> bearerToken,
+          HttpHeaderNames.Accept        -> "application/vnd.hmrc.1.0+json",
```
Comment (TheDanielDoyle) [inline src/test/scala/uk/gov/hmrc/perftests/uknwauthcheckerapi/requests/UknwAuthCheckerApiRequests.scala]: Can you move both of these string literals into vals, to cut down reuse.

File: src/test/scala/uk/gov/hmrc/perftests/uknwauthcheckerapi/util/Helper.scala [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+
+  def withFileAsString(fileName: String): String = {
+    val source = Source.fromFile(fileName, "utf-8")
+    try
+      source.mkString
+    finally
+      source.close()
+  }
+
+  val singleEoriJsonBody: String           = withFileAsString("src/test/resources/data/singleEori.json")
```
Comment (TheDanielDoyle) [inline src/test/scala/uk/gov/hmrc/perftests/uknwauthcheckerapi/util/Helper.scala]: Please run scalafmtAll

File: src/test/scala/uk/gov/hmrc/perftests/uknwauthcheckerapi/requests/UknwAuthCheckerApiRequests.scala (hunk 6 above)
Comment (TheDanielDoyle) [reply src/test/scala/uk/gov/hmrc/perftests/uknwauthcheckerapi/requests/UknwAuthCheckerApiRequests.scala]: Also, the logic here is strange. Wouldn't you need **Bearer ${accessToken}** in both scenarios?

File: src/test/resources/services.conf (hunk 3 above)
Comment (schemabotview) [reply src/test/resources/services.conf]: Replaced with given url.

File: src/test/resources/data/500Eori.json (hunk 4 above)
Comment (schemabotview) [reply src/test/resources/data/500Eori.json]: Formatted the json.

File: src/test/resources/data/100Eori.json (hunk 5 above)
Comment (schemabotview) [reply src/test/resources/data/100Eori.json]: Formatted the json.

File: src/test/scala/uk/gov/hmrc/perftests/uknwauthcheckerapi/requests/UknwAuthCheckerApiRequests.scala (hunk 6 above)
Comment (schemabotview) [reply src/test/scala/uk/gov/hmrc/perftests/uknwauthcheckerapi/requests/UknwAuthCheckerApiRequests.scala]: yes, I put $$ as it contains reference variable in turn.
and there 2 cases one with itself and with Bearer prefix.

File: src/test/scala/uk/gov/hmrc/perftests/uknwauthcheckerapi/requests/UknwAuthCheckerApiRequests.scala (hunk 7 above)
Comment (schemabotview) [reply src/test/scala/uk/gov/hmrc/perftests/uknwauthcheckerapi/requests/UknwAuthCheckerApiRequests.scala]: Exported as local value.

File: src/test/scala/uk/gov/hmrc/perftests/uknwauthcheckerapi/util/Helper.scala (hunk 8 above)
Comment (schemabotview) [reply src/test/scala/uk/gov/hmrc/perftests/uknwauthcheckerapi/util/Helper.scala]: formatted the code now.

File: src/test/resources/data/1000Eori.json [hunk 9]
```diff
@@ -0,0 +1,1006 @@
+{
+  "date": "2024-02-08",
+  "authType": "UKNW",
+  "eoris": [
+    "GB769039790992536",
```
Comment (TheDanielDoyle) [inline src/test/resources/data/1000Eori.json]: These all contain the same number. It's not really a reflection of a valid request.

File: src/test/scala/uk/gov/hmrc/perftests/uknwauthcheckerapi/requests/UknwAuthCheckerApiRequests.scala [hunk 10]
```diff
@@ -0,0 +1,98 @@
...
+  private val acceptType          = "application/vnd.hmrc.1.0+json"
+  private val mimeType            = "application/json"
+  private val bearerToken: String = if (runLocal) s"$${accessToken}" else s"Bearer $${accessToken}"
+
+  val postAuthorisation: HttpRequestBuilder =
+    http("Post Authorisations")
+      .post(s"$baseUrl$route": String)
+      .body(StringBody(Helper.singleEoriJsonBody))
+      .headers(
+        Map(
```
Comment (TheDanielDoyle) [inline src/test/scala/uk/gov/hmrc/perftests/uknwauthcheckerapi/requests/UknwAuthCheckerApiRequests.scala:39]: Extract this into a function to reduce code duplication.

Comment (schemabotview) [reply src/test/scala/uk/gov/hmrc/perftests/uknwauthcheckerapi/requests/UknwAuthCheckerApiRequests.scala:39]: Extracted to a method

File: src/test/resources/data/1000Eori.json (hunk 9 above)
Comment (schemabotview) [reply src/test/resources/data/1000Eori.json]: Changed the requests to contain different EORI

File: src/test/resources/journeys.conf (hunk 2 above)
Comment (schemabotview) [reply src/test/resources/journeys.conf]: Changed load as mentioned in documentation.

**Review by TheDanielDoyle — APPROVED** [review]

**Pull request merged**